remill/Arch/AArch64/Arch.cpp

```diff
@@ -4045,6 +4106,174 @@ bool TryDecodeSCVTF_D64_FLOAT2INT(const InstData &data, Instruction &inst) {
+bool TryDecodeBIC_ASIMDSAME_ONLY(const InstData &data, Instruction &inst) {
+  return TryDecodeORR_ASIMDSAME_ONLY(data, inst);
+}
+
+// EOR  <Vd>.<T>, <Vn>.<T>, <Vm>.<T>
+bool TryDecodeEOR_ASIMDSAME_ONLY(const InstData &data, Instruction &inst) {
+  return TryDecodeORR_ASIMDSAME_ONLY(data, inst);
+}
+
+// BIT  <Vd>.<T>, <Vn>.<T>, <Vm>.<T>
+bool TryDecodeBIT_ASIMDSAME_ONLY(const InstData &data, Instruction &inst) {
+  AddQArrangementSpecifier(data, inst, "16B", "8B");
+  AddRegOperand(inst, kActionReadWrite, kRegV, kUseAsValue, data.Rd);
+  AddRegOperand(inst, kActionRead, kRegV, kUseAsValue, data.Rn);
+  AddRegOperand(inst, kActionRead, kRegV, kUseAsValue, data.Rm);
+  return true;
+}
+
+// BIF  <Vd>.<T>, <Vn>.<T>, <Vm>.<T>
+bool TryDecodeBIF_ASIMDSAME_ONLY(const InstData &data, Instruction &inst) {
+  return TryDecodeBIT_ASIMDSAME_ONLY(data, inst);
+}
+
+// ADDV  <V><d>, <Vn>.<T>
+bool TryDecodeADDV_ASIMDALL_ONLY(const InstData &data, Instruction &inst) {
+  if (data.size == 0x2 && !data.Q) {
+    return false;  // `if size:Q == '100' then ReservedValue();`
+  } else if (data.size == 0x3) {
+    return false;  // `if size == '11' then ReservedValue();`.
+  }
+  const uint64_t esize = 8ULL << data.size;
+  const uint64_t datasize = data.Q ? 128 : 64;
+  AddArrangementSpecifier(inst, datasize, esize);
+  AddRegOperand(inst, kActionWrite, kRegV, kUseAsValue, data.Rd);
+  AddRegOperand(inst, kActionRead, kRegV, kUseAsValue, data.Rn);
+  return true;
+}
+
+// UMINV  <V><d>, <Vn>.<T>
+bool TryDecodeUMINV_ASIMDALL_ONLY(const InstData &data, Instruction &inst) {
+  return TryDecodeADDV_ASIMDALL_ONLY(data, inst);
+}
+
+// UMAXV  <V><d>, <Vn>.<T>
+bool TryDecodeUMAXV_ASIMDALL_ONLY(const InstData &data, Instruction &inst) {
+  return TryDecodeADDV_ASIMDALL_ONLY(data, inst);
+}
+
+// SMAXV  <V><d>, <Vn>.<T>
+bool TryDecodeSMAXV_ASIMDALL_ONLY(const InstData &data, Instruction &inst) {
+  return TryDecodeADDV_ASIMDALL_ONLY(data, inst);
+}
+
+// SMINV  <V><d>, <Vn>.<T>
+bool TryDecodeSMINV_ASIMDALL_ONLY(const InstData &data, Instruction &inst) {
+  return TryDecodeADDV_ASIMDALL_ONLY(data, inst);
+}
+
+// FMAXV  <V><d>, <Vn>.<T>
+bool TryDecodeFMAXV_ASIMDALL_ONLY_H(const InstData &data, Instruction &inst) {
+  AddQArrangementSpecifier(data, inst, "8H", "4H");
+  AddRegOperand(inst, kActionWrite, kRegV, kUseAsValue, data.Rd);
+  AddRegOperand(inst, kActionRead, kRegV, kUseAsValue, data.Rn);
+  return true;
+}
+
+// FMAXV  <V><d>, <Vn>.<T>
+bool TryDecodeFMAXV_ASIMDALL_ONLY_SD(const InstData &data, Instruction &inst) {
+  if (!(!data.sz && data.Q)) {
+    return false;  // `if sz:Q != '01' then ReservedValue();`
+  }
+  inst.function += "_4S";
+  AddRegOperand(inst, kActionWrite, kRegV, kUseAsValue, data.Rd);
+  AddRegOperand(inst, kActionRead, kRegV, kUseAsValue, data.Rn);
+  return true;
+}
+
+// FMINNMV  <V><d>, <Vn>.<T>
+bool TryDecodeFMINNMV_ASIMDALL_ONLY_H(const InstData &data, Instruction &inst) {
+  return TryDecodeFMAXV_ASIMDALL_ONLY_H(data, inst);
```

Comment:
These decoders have non-generic names but then they're used generically. What I mean is, e.g., `TryDecodeFMAX` being used to decode `TryDecodeFMIN`. It looked like an error at first but I realize this is just code re-use.

Would it be less error-prone if you always had generically named decoders like `TryDecodeFxxxASIMDALL_ONLY_H`? Not worth it?

Reply:
It's probably worth is as a rule going forward, but it's more busywork for me now. Filed as an enhancement Issue #171.